File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int> set;

        for(int i = 0; i<nums.size(); i++){
            if(!set.count(nums[i])){
                set.insert(nums[i]);
            }else{
                return true;
            }
        }

        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.go
func containsDuplicate(nums []int) bool {
    mapD := make(map[int]bool) // map to track duplicate numbers

    for _, num := range nums {
        if _, exists := mapD[num]; exists{
            return true;
        } else {
            mapD[num] = true;
        }
    }


    return false;
}